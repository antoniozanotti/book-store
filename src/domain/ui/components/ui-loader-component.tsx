import { UiLoaderComponentInterface } from "./ui-loader-component-interface";
import { Loader2Icon } from "lucide-react";

export function UiLoaderComponent({ ...props }: UiLoaderComponentInterface) {
  const iconStyles = "animate-spin";
  props.className = props.className
    ? `${iconStyles} ${props.className}`
    : iconStyles;

  return <Loader2Icon {...props} />;
}
