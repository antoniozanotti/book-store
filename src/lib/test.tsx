import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
