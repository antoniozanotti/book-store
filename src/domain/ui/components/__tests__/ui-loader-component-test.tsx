import { render, screen } from "@/lib/test";
import { UiLoaderComponent } from "../ui-loader-component";

describe("UiLoaderComponent", () => {
  it("should render", async () => {
    const { container } = render(<UiLoaderComponent />);
    expect(container).toMatchSnapshot();
  });
});