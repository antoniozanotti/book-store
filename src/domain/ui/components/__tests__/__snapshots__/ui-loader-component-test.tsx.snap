// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiLoaderComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <svg
      class="lucide lucide-loader-circle animate-spin"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12a9 9 0 1 1-6.219-8.56"
      />
    </svg>
  </div>
</div>
`;
